src/standard/utils.html

```diff
@@ -367,50 +367,52 @@
-      link.rel = 'import';
-      link.href = href;
-      var list = Polymer.Base.importHref.imported = 
-        Polymer.Base.importHref.imported || {};
-      var cached = list[link.href];
-      var imprt = cached || link;
+      var link = document.head.querySelector('[href="' + href + '"]');
       var self = this;
-      var loadListener = function(e) {
-        e.target.__firedLoad = true;
-        e.target.removeEventListener('load', loadListener);
-        e.target.removeEventListener('error', errorListener);
-        return onload.call(self, e);
-      };
-      var errorListener = function(e) {
-        e.target.__firedError = true;
-        e.target.removeEventListener('load', loadListener);
-        e.target.removeEventListener('error', errorListener);
-        return onerror.call(self, e);
-      };
-      if (onload) {
-        imprt.addEventListener('load', loadListener);
+
+      if (link == null) {
+        link = document.createElement('link');
+        link.rel = 'import';
+        link.__dynamicImport = true;
       }
-      if (onerror) {
-        imprt.addEventListener('error', errorListener);
+
+      if (optAsync) {
+        link.setAttribute('async', '');
       }
-      // if already loaded/erroed, fire 'fake' load/error event
-      if (cached) {
-        if (cached.__firedLoad) {
-          cached.dispatchEvent(new Event('load'));
-        }
-        if (cached.__firedError) {
-          cached.dispatchEvent(new Event('error'));
+
+      var loadListener = function(event) {
+        link.removeEventListener('load', loadListener);
+        link.removeEventListener('error', errorListener);
+        // In case of a successful load, cache the load event on the link so
+        // that it can be used to short-circuit this method in the future when
+        // it is called with the same href param.
+        link.__dynamicImportLoaded = true;
+
+        if (onload != null) {
+          return onload.call(self, event);
         }
-      // otherwise put in dom!
-      } else {
-        list[link.href] = link;
-        optAsync = Boolean(optAsync);
-        if (optAsync) {
-          link.setAttribute('async', '');
+      };
+      var errorListener = function(event) {
+        link.removeEventListener('load', loadListener);
+        link.removeEventListener('error', errorListener);
+        // In case of an error, remove the link from the document so that it
+        // will be automatically created again the next time `importHref` is
+        // called.
+        document.head.removeChild(link);
+        if (onerror != null) {
+          return onerror.call(self, event);
         }
+      };
+
+      link.addEventListener('load', loadListener);
+      link.addEventListener('error', errorListener);
+
+      if (link.parentNode == null) {
+        link.href = href;
         document.head.appendChild(link);
+      } else if (link.__dynamicImportLoaded || !link.__dynamicImport) {
```

Comment:
Rationale here is that imports that were loaded dynamically and finished, or weren't loaded dynamically to begin with, should fire a synthetic `load` event. This might fall apart for imports that were loaded outside of `importHref`, but aren't fully loaded yet by the time we get here. Thoughts?